File: 1922. Count Good Numbers.java
class Solution {
    long mod = 1000000007;
    public int countGoodNumbers(long n) {
        return (int) ((expo(5, (n + 1)/2) * expo(4, n/2)) % mod);
    }
    public long expo(int x, long y){
        long res = 1;
        long mul = x;
        while(y > 0){
            if(y % 2 == 1){
                res = (res * mul) % mod;
            }
            mul = (mul * mul) % mod;
            y/=2;
        }
        return res;
    }
}

File: 1922. Count Good Numbers.cpp
class Solution {
    static long MOD = (long)1e9 + 7;
    public long fastPow(long n, long pow){
        if(pow == 0) return 1;
        else if(pow % 2 == 0){
            long res = fastPow(n, pow / 2) % MOD;
            return (res * res) % MOD;
        }
        else{
            return (n * (fastPow(n, pow - 1))) % MOD;
        }
    }
    public int countGoodNumbers(long n) {
        long even = (n + 1)/2;
        long odd = n / 2;
        
        long part1 = fastPow(5, even) % MOD;
        long part2 = fastPow(4, odd) % MOD;

        return (int)((part1 * part2) % MOD);
    }
}